osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringConfigurationEndpoint.java

```diff
@@ -205,18 +208,18 @@ public GetFirmwareVersionGasAsyncResponse getFirmwareVersionGas(
 
         return (GetFirmwareVersionGasAsyncResponse) this.handleGetFirmwareVersion(organisationIdentification,
                 deviceIdentification, new GetFirmwareVersionQuery(true), messagePriority, scheduleTime, responseUrl,
-                response);
+                response, Boolean.parseBoolean(bypassRetry));
     }
 
     private AsyncResponse handleGetFirmwareVersion(final String organisationIdentification,
```

Comment:
Consider refactoring by moving 
```final GetFirmwareVersionGasAsyncResponse response = new GetFirmwareVersionGasAsyncResponse();```
into the method instead of passing as parameter and by
updating the method to return a `GetFirmwareVersionGasAsyncResponse` instead of a `AsyncResponse`.

Reply:
Changed